clients/roscpp/src/libros/service_server_link.cpp

```diff
@@ -198,6 +198,12 @@ void ServiceServerLink::onResponseOkAndLength(const ConnectionPtr& conn, const b
 
   if (len > 1000000000)
   {
+    {
+      boost::mutex::scoped_lock lock(call_queue_mutex_);
```

Comment:
isn't it `finished_mutex_` to protect `finished_` and `finished_condition_`?

Reply:
It's just my test here, please ignore it.